balloc/balloc.c

```diff
@@ -710,9 +796,7 @@ static void balloc_group_write_do(struct m0_be_tx_bulk *tb,
 	m0_ext_init(&ext);
 	balloc_debug_dump_extent("create...", &ext);
 
-	key = (struct m0_buf)M0_BUF_INIT_PTR(&ext.e_end);
-	val = (struct m0_buf)M0_BUF_INIT_PTR(&ext.e_start);
-	rc = btree_insert_sync(&bal->cb_db_group_extents, tx, &key, &val);
+	rc = balloc_ext_insert(bal->cb_db_group_extents, tx, ext);
```

Comment:
Balloc extent tree has free extents as records. The record has extend end as key and extend start as value. The extents needs to be arranged in increasing order in the extent balloc btree. To have extents in increasing order, the keys (i.e. extent end) needs to be in increasing order.  Because of the memcmp limitation of node find, the keys needs to be converted to big endian format. Hence extents are passed by value to balloc_ext_insert() function and in balloc_ext_insert() , extent end is converted to bigendian. The balloc_ext_insert() function then calls btree_insert_sync().

Morever balloc_ext_insert() is inline function and it eventually calls btree_insert_sync().